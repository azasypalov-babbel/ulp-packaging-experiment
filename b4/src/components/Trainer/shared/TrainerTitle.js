import React from 'react';
import PropTypes from 'prop-types';

import { rem } from 'polished';
import styled from 'styled-components';
import { markupStringToHTML } from '@lessonnine/babbel-markup-helper.js';

import Title from '../../shared/Title';

const StyledTitle = styled(Title)`
  margin-left: 2.25rem;
  margin-right: 2.25rem;
  margin-bottom: ${rem(24)};

  @media (min-width: ${(props) => props.theme.viewports.breakpoints.xxsmall}) {
    margin-bottom: ${rem(32)};
  }

  @media (min-width: ${(props) => props.theme.viewports.breakpoints.xsmall}) {
    margin-bottom: ${rem(48)};
  }
`;

const TrainerTitle = ({ text, ...props }) => (
  <StyledTitle
    data-selector="title"
    textAlign="center"
    fontFamily="fontMilliardSemi"
    color="spaceGray"
    fontSize="big"
    as="h1"
    dangerouslySetInnerHTML={{ __html: markupStringToHTML(text) }}
    {...props}
  />
);

TrainerTitle.propTypes = {
  text: PropTypes.string.isRequired
};

export default TrainerTitle;
