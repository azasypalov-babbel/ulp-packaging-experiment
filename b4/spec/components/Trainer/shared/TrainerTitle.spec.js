import React from 'react';
import { shallow } from 'enzyme';

import TrainerTitle from '../../../../src/components/Trainer/shared/TrainerTitle';

import { markupStringToHTML } from '@lessonnine/babbel-markup-helper.js';
jest.mock('@lessonnine/babbel-markup-helper.js', () => ({
  markupStringToHTML: jest.fn((a) => a)
}));

const defaultProps = {
  text: 'formatt{ed}'
};

describe('<TrainerTitle />', () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <TrainerTitle {...defaultProps} />
    );
  });

  test('it renders', () => {
    expect(component).toMatchSnapshot();
  });

  test('calls babbel markup helper method', () => {
    expect(markupStringToHTML).toHaveBeenCalled();
  });
});
