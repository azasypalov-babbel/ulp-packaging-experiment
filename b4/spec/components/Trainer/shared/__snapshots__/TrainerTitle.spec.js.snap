// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainerTitle /> it renders 1`] = `
<Styled(Title)
  as="h1"
  color="spaceGray"
  dangerouslySetInnerHTML={
    Object {
      "__html": "formatt{ed}",
    }
  }
  data-selector="title"
  fontFamily="fontMilliardSemi"
  fontSize="big"
  textAlign="center"
/>
`;
